netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -365,10 +437,22 @@ public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
        * lifetime and we only want to configure it once.
        */
       if (handlers != null) {
-        ctx.pipeline().addFirst(handlers);
+        for (ChannelHandler handler : handlers) {
```

Comment:
Previously there was only one negotiator. Now there can be multiple. addFirst assumed it was the only thing in the channel pipeline. Now we make sure to put the other handlers immediately before this handler, which causes nesting to work correctly.